crates/build/src/crate_metadata.rs

```diff
@@ -66,34 +66,46 @@ impl CrateMetadata {
         let (metadata, root_package) = get_cargo_metadata(manifest_path)?;
         let mut target_directory = metadata.target_directory.as_path().join("ink");
 
-        // Normalize the package and lib name.
+        // Normalize the package name.
         let package_name = root_package.name.replace('-', "_");
-        let lib_name = &root_package
+
+        if let Some(lib_name) = &root_package
             .targets
             .iter()
             .find(|target| target.kind.iter().any(|t| t == "cdylib"))
```

Comment:
This happens after we added it, right?

Reply:
Please elaborate.

We are using the package name now, the `lib_name` is just replaced by a check to show a warning in case anyone is depending on the name of the artifact being the same as the `lib` name.

Reply:
We search for a lib using `cdylib`.  But it is not necessarily in the source manifest. I am asking if it happens after we modified it with:
```
.with_added_crate_type("cdylib")?
```

Reply:
Ah gotcha. No this all happens before any modifications are made. It is about reading the metadata for the contract project.

The heuristic previously was assuming a `cdylib` target. Now it only keeps that in order to warn users if they are still relying on it.

Reply:
But they could just add a name entry there without adding `cdylib`. Then this error message would not be emitted. We basically just need to match any `[lib]` table. There can only be one per package anyways.

Reply:
True. If they did that previously though the name is ignored because of the `cdylib` predicate, and the package name will be used. The thinking was to only produce the warning in the case when the lib name would have been used for the name previously. Hence the additional check for whether `package_name != lib_name`.